build.sh

```diff
@@ -0,0 +1,138 @@
+                #Install pip using APT
+                sudo apt-get install -y python3-pip > /dev/null
+            elif [ $ID == 'fedora' ] || [ $ID_LIKE == 'rhel fedora' ]; then
+                #Request sudo privilleges
+                requestSudo
+                #Install pip using DNF
+                sudo dnf install -y python3-pip > /dev/null
+            elif [ $ID_LIKE == 'arch' ] || [ $ID_LIKE == 'archlinux' ]; then
+                #Request sudo privilleges
+                requestSudo
+                #Install pip using pacman
+                sudo pacman -S --noconfirm python-pip > /dev/null
+            else
+                #Distro-agnostic way to install pip
+                #Official documentation: https://pip.pypa.io/en/stable/installation/
+                printf "%b %bYour distribution is not supported!%b\\n" "${CROSS}" "${COL_LIGHT_RED}" "${COL_NC}"
+                printf "%b %bInstalling pip via alternative method...%b\\n" "${INFO}"
+
+                #Download get-pip.py only if it doesn't exist
+                if ! [ -f tmp/get-pip.py ]; then
+                    wget -q --show-progress -P tmp/ https://bootstrap.pypa.io/get-pip.py
+                fi
+
+                #Install pip using the official script
+                python3 tmp/get-pip.py
+            fi
+
+            #Verify that pip has been installed
+            if hash pip3 2>/dev/null; then
+                printf "%b %bDone!%b\\n" "${INFO}"
+            else
+                printf "%b %bFailed to install pip.%b\\n" "${CROSS}" "${COL_LIGHT_RED}" "${COL_NC}"
+                exit 22;
+            fi
+        else
+            printf "%b %bOperation cancelled by the user. Aborting.%b\\n" "${CROSS}" "${COL_LIGHT_RED}" "${COL_NC}"
+            exit 125;
+        fi
+    fi
+
+    #Print pip version
+    printf "%b %bpip $(pip3 -V | awk '{print $2;}')%b\\n" "${TICK}" "${COL_LIGHT_GREEN}" "${COL_NC}"
+
+    #Install dev dependencies
+    printf "%b %bInstalling dependencies...%b\\n" "${INFO}"
+    pip3 install -r requirements.rc > /dev/null
+
+    #Check if pyinstaller is installed
+    if ! hash pyinstaller 2>/dev/null; then
+        printf "%b %bRestart your computer for the changes to be applied.!%b\\n" "${CROSS}" "${COL_LIGHT_RED}" "${COL_NC}"
+        exit 22
+    fi
+
+    #Make sure tmp does not already exist
+    rm -rf tmp
+    #Create a directory for storing temporary files
+    mkdir tmp
+    #Create an empty file so that environment setup does not happen again
+    touch tmp/.environment_setup_complete
+
+    printf "%b %bDone installing dependencies!%b\\n" "${TICK}" "${COL_LIGHT_GREEN}" "${COL_NC}"
+fi
+
+printf "%b %bBuilding package...%b\\n" "${INFO}" "${COL_LIGHT_YELLOW}"
+
+#Build the package, using the official instructions
+pyinstaller main.pyw --onefile --clean --noconfirm \
+    --version-file "version.rc" \
+    --add-data "data/*:data" \
+    --name "imd-gui" \
+    --icon "data/jukebox_256.ico" \
+    --distpath bin 2>tmp/latest.log \
```

Comment:
I think that the binary file should work on all popular distributions as that is the case most of the time. I can't imagine anyone trying to run it on an obscure distro from 2013. Right?

- Yes, I think it does. At least from what I can see there is only one file in build/ regardless of how many times I have run the script.
- The initial idea was to pipe everything to a file to make the output look cleaner. Is your suggestion to make everything visible or only the errors? It could point the user to just open the log file in case of any error.
- It is a good idea to change the directory.

Reply:
On Manjaro which i am on the binary works perfectly!